esp/public/media/scripts/program/modules/modulequestions.js

```diff
@@ -0,0 +1,65 @@
+
+$j(document).ready(function(){
+    // when the page loads, check/select the questions induced by the modules
+    // that are already selected (if any, from a template program e.g.)
+    $j(".hidden-field").parents("tr").hide()
+    modulesToQuestions();
+    // questionsToModules is called when the form gets submitted, but the following
+    // is useful for debugging
+    /*
+    $j("input[name='program_module_questions']" ).change(function() {
+        questionsToModules();
+    });*/
+});
+
+
+function questionsToModules() {
+    var nQuestions = $j("#id_program_module_questions").children().length;
+    var modules = [];
+    for (var i=0; i < nQuestions; i++) {
+        // record which questions are checked (and save their corresponding modules)
+        if ($j("#id_program_module_questions_" + String(i)).prop("checked"))
+            $j.merge(modules, $j("#id_program_module_questions_"+i).val().split(","))
+    }
+    // Now just check those modules in the list.
+    modules = $j.unique(modules)
+    modules.sort() // $j.uniqueSort doesn't exist for jQuery < 2.2
+    $j("#id_program_modules").val(modules);
+    return
+};
+
+
+// Given the list of selected modules in the program_modules MultipleChoiceField,
+// check the checkboxes in the program modules questions list that they induce
+// This is hard-coded based on our knowledge of modules and needs to be
+// updated as new modules are introduced.
+function modulesToQuestions() {
+    // the IDs of the modules currently selected
+    var currentlySelectedIds = $j("select#id_program_modules").val();
+    // if there are none, skip the rest of this
+    if (currentlySelectedIds == null || currentlySelectedIds.length == 0)
+        return
+    // otherwise, get the questions and see whether we should check them
+    var questions = $j("#id_program_module_questions").children()
+    // check if all the modules associated with a given question are selected
```

Comment:
If we are editing the settings for a program that was setup before this change (or was setup manually for some reason), if someone had enabled only some of the modules associated with a question (again, for some weird reason), wouldn't this cause that module to get disabled upon submitting the form?

Reply:
Same with if we are using a program from before as a template?

Reply:
Yes, they would. I'm having trouble thinking of scenarios where you would only want some of the questions. I think it's more likely that someone configured a previous program in a messy way and this would clean that up for them. But if you really, really want only some, then you can go back and add them manually through the admin panel.

Reply:
Oh, I agree we should clean it up for them. But here, don't we remove modules when all of the modules for a question aren't satisfied instead of adding the ones they are missing?

Reply:
only if they don't check the question. If they do check it, then we add the ones that aren't satisfied.

Reply:
Right, but if someone is using a program as a template, it won't check the question box unless all of the modules for the questions are included for the template program. Then someone will submit the form assuming the modules will be the same as the template program, but they will actually end up with fewer modules than the template programs because the modules that were enabled for those boxes that aren't checked will be NOT enabled.

Reply:
I guess I just thought it was better to under-include than over-include if they didn't have all of the right modules selected. @kkbrum?

Reply:
That's fine, but if you are using a program as a template I think you should ALWAYS have AT LEAST all of those modules in your new program (provided you aren't changing anything).

Reply:
OK, I propose a compromise (because after thinking about it, either way could have unwanted consequences): if only some of the modules for a question are included, don't check the box, but at least indicate this discrepancy by coloring the background of that question or something.

Reply:
You win :) 1a0c2dd